packages/react/src/dropdown-menu/DropdownMenuItemLink.tsx

```diff
@@ -4,7 +4,9 @@ import { LinkProps, useLinkComponent } from '../core';
 import { IconProps } from '../icon';
 import { DropdownMenuItem } from './DropdownMenuItem';
 
-export type DropdownMenuItemLinkProps = LinkProps & {
+export type DropdownMenuItemLinkProps = Omit<LinkProps, 'onClick'> & {
+	/** Function to be fired following a click event of the item. */
+	onClick?: () => void;
```

Comment:
This fixes a small typescript issue, with the addition of `onClick` in `LinkProps`